### PR TITLE
Details page

### PR DESCRIPTION
**Pull request by MarcelSchueren**
Würde mich über ein kurzes Review sehr freuen :)

### CODE REVIEW

**Review by macben888 — APPROVED** [review]

File: backend/src/main/java/de/neufische/mydepot/api/PortfolioItemApiService.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+public class PortfolioItemApiService {
+
+    private final YahooFinanceService yahooFinanceService;
+
+    @Autowired
+    public PortfolioItemApiService(YahooFinanceService yahooFinanceService) {
+        this.yahooFinanceService = yahooFinanceService;
+    }
+
+    public PortfolioItem getPortfolioItemBySymbol(String symbol) {
```
[the remark this answers is not in the document]
Comment (MarcelSchueren) [reply backend/src/main/java/de/neufische/mydepot/api/PortfolioItemApiService.java:23]: Meant is a "ticker-symbol", a symbol to find the correct stock (similar to the German "Wertpapierkennnummer"

File: backend/src/main/java/de/neufische/mydepot/api/YahooFinanceService.java [hunk 2]
```diff
@@ -15,20 +17,88 @@
...
+    }
+
+    public BigDecimal getRegularMarketPrice(Stock stock) {
+        if (stock.getCurrency().equals("USD")) {
+            return changeUSDinEUR(stock.getQuote().getPrice());
+        }
+        return stock.getQuote().getPrice();
+    }
+
+    private BigDecimal changeUSDinEUR(BigDecimal priceInUSD) {
```
[the remark this answers is not in the document]
Comment (MarcelSchueren) [reply backend/src/main/java/de/neufische/mydepot/api/YahooFinanceService.java:62]: Thanks, I will consider to add this functionality.

File: frontend/src/components/Portfolio.js [hunk 3]
```diff
@@ -1,26 +1,38 @@
...
-                <Typography variant="h5">{depot.valueOfPortfolio.toFixed(2)} $ </Typography>
-                   </CardContent>
-            <CardActions style={{justifyContent: 'center'}}>
-                <Button size="small" onClick={handleClick}>Open</Button>
-            </CardActions>
+            <Typography className={classes.cardDepotName}>{depot.name}</Typography>
+            <Typography className={classes.cardValueOfPortfolio}>{depot.valueOfPortfolio.toFixed(2)} € </Typography>
+            <Typography className={classes.cardPortfolioPercentageChange}
+                        color={percentageChange < 0 ? theme.palette.error.light : "lightgreen"}>
+                {percentageChange > 0 ? '+' : <></>}
```
[the remark this answers is not in the document]
Comment (MarcelSchueren) [reply frontend/src/components/Portfolio.js:25]: Thanks!

File: frontend/src/components/Portfolio.js [hunk 4]
```diff
@@ -1,26 +1,38 @@
...
+                {percentageChange.toFixed(2)}
+                %
+            </Typography>
+            <Typography className={classes.cardPortfolioAbsoluteChange}
+                        color={absoluteGain < 0 ? theme.palette.error.light : "lightgreen"}>
+                {percentageChange > 0 ? '+' : <></>}
```
[the remark this answers is not in the document]
Comment (MarcelSchueren) [reply frontend/src/components/Portfolio.js:31]: Thanks, I changed this.

File: frontend/src/components/PortfolioItem.js [hunk 5]
```diff
@@ -1,24 +1,53 @@
...
+import theme from "../styling/theme";
+import {useHistory} from "react-router-dom";
 
-export default function PortfolioItem({stock}) {
+export default function PortfolioItem({stock, openPortfolioItem, short}) {
 
     const classes = useStyles()
+    const history = useHistory()
+
+    const absoluteChange = -(stock.quantity * stock.boughtAtPricePerShare
```
[the remark this answers is not in the document]
Comment (MarcelSchueren) [reply frontend/src/components/PortfolioItem.js:11]: changed :)

File: frontend/src/pages/NewDepotPage.js [hunk 6]
```diff
@@ -1,15 +1,114 @@
...
+        }
+        getPortfolioItem(symbol)
+            .catch(() => {
+                setSymbolIsWrong(true)
+            })
+            .then(portfolioItem => setActivePortfolioItem(portfolioItem))
+    }
+
+    const saveDepot = () => {
+        if (portfolioName === '' || portfolioName === undefined) {
```
[the remark this answers is not in the document]
Comment (MarcelSchueren) [reply frontend/src/pages/NewDepotPage.js:42]: Thanks! I've changed, but what does the '?' mean?

**Pull request merged**